packages/components/src/components/modal/modal.js

```diff
@@ -104,10 +104,18 @@ class Modal extends mixin(
         this.element.offsetHeight > 0
       ) {
         this.previouslyFocusedNode = this.element.ownerDocument.activeElement;
-        (
+        const focusableItem =
           this.element.querySelector(this.options.selectorPrimaryFocus) ||
-          this.element.querySelector(settings.selectorTabbable)
-        ).focus();
+          this.element.querySelector(settings.selectorTabbable);
+        focusableItem.focus();
+        if (__DEV__ && focusableItem === null) {
```

Comment:
The 1st argument is `warning()` is like C `assert()`. That said, you don't need `focusableItem === null` here.